packages/components/src/sectorDetailsWrapper/SectorDetailsWrapper.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import * as React from 'react';
+import { Touchable, Icon, StyleSheet } from '@kiwicom/universal-components';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { Animated } from 'react-native';
+
+type Props = {|
+  +children: React.Node,
+|};
+
+type State = {|
+  +expandSegments: boolean,
+  +isDisabled: boolean,
+|};
+
+export default class SectorDetailsWrapper extends React.Component<
+  Props,
+  State,
+> {
+  constructor() {
+    super();
+
+    this.state = {
+      expandSegments: false,
+      isDisabled: false,
+    };
+
+    this.spinValue = new Animated.Value(0);
+  }
+
+  spinValue: Animated.Value;
+
+  toggleExpanded = () => {
+    this.setState({ isDisabled: true }, () => {
+      Animated.timing(this.spinValue, {
+        toValue: this.state.expandSegments ? 0 : 1,
+        duration: 250,
+        useNativeDriver: true,
+      }).start(() => {
+        this.setState(state => ({
+          expandSegments: !state.expandSegments,
+          isDisabled: false,
+        }));
+      });
+    });
+  };
+
+  render() {
+    const spin = this.spinValue.interpolate({
+      inputRange: [0, 1],
+      outputRange: ['0deg', '180deg'],
+    });
+
+    return (
+      <>
+        <Touchable
+          onPress={this.toggleExpanded}
+          disabled={this.state.isDisabled}
+        >
+          <Animated.View
+            style={[
+              styles.iconWrapper,
+              {
+                transform: [{ rotate: spin }, { perspective: 1000 }],
+              },
+            ]}
+          >
+            <Icon
```

Comment:
Don't we have the same animated icon in BookingDetail?

Reply:
This is the one. Maybe the summary is not that clear. But I extracted this whole wrapper from there and made it reusable.

Reply:
I see, in that case it's perfect :)